src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -118,28 +121,32 @@ public async ValueTask SynchronizeProjectAssetsAsync(ProjectStateChecksums proje
 
         using (Logger.LogBlock(FunctionId.AssetService_SynchronizeAssetsAsync, Checksum.GetChecksumsLogInfo, checksums, cancellationToken))
         {
-            using var _1 = ArrayBuilder<Checksum>.GetInstance(checksums.Count, out var missingChecksums);
+            var missingChecksums = s_checksumListPool.Allocate();
```

Comment:
can you use .GetPooledObject() instead?  then you can use this in a `using` and be sure to return this even if canceled.

Reply:
I believe that would called the pooledobjject's releaser, which will do the size threshold check.

Reply:
Ooh, there is a trimOnFree param, let me try that out

Reply:
Works wonderfully, thanks!